src/content/app/genome-browser/hooks/useGenomeBrowserAnalytics.ts

```diff
@@ -0,0 +1,303 @@
+  ) => {
+    sendTrackEvent({
+      category: 'track_settings',
+      label: selectedCog,
+      action: 'transcript_labels_' + (shouldShowTranscriptIds ? 'on' : 'off')
+    });
+  };
+
+  const trackApplyToAllInTackSettings = (
+    selectedCog: string,
+    shouldApplyToAll: boolean
+  ) => {
+    sendTrackEvent({
+      category: 'track_settings',
+      label: selectedCog,
+      action: 'apply_to_all - ' + (shouldApplyToAll ? 'unselected' : 'selected')
+    });
+  };
+
+  const trackTrackPanelTabChange = (selectedTrackPanelTab: TrackSet) => {
+    sendTrackEvent({
+      category: 'track_panel_tab',
+      label: selectedTrackPanelTab,
+      action: 'selected'
+    });
+  };
+
+  const trackTrackPanelSectionToggled = (sectionName: string) => {
+    sendTrackEvent({
+      category: 'track_panel',
+      label: sectionName,
+      action: 'section_toggled'
+    });
+  };
+
+  const trackResetIcon = () => {
+    sendTrackEvent({
+      category: 'reset_icon',
+      action: 'clicked'
+    });
+  };
+
+  const trackSidebarModalViewToggle = (view: string) => {
+    sendTrackEvent({
+      category: 'sidebar_toolstrip',
+      action: 'modal_opened',
+      label: view
+    });
+  };
+
+  const trackDrawerOpened = (drawerView: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'opened',
+      label: drawerView
+    });
+  };
+
+  const trackDrawerSequenceViewed = (sequenceType: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'sequence_viewed',
+      label: sequenceType
+    });
+  };
+
+  const trackDrawerSequenceCopied = (sequenceType: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'sequence_copied',
+      label: sequenceType
+    });
+  };
+
+  const trackDrawerSequenceDownloaded = (selectedOptions: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'sequence_downloaded',
+      label: selectedOptions
+    });
+  };
+
+  const trackSpeciesChangeLinkClicked = () => {
+    sendTrackEvent({
+      category: 'app_bar',
+      action: 'change_link_clicked'
+    });
+  };
+
+  const trackAppBarGenomeChanged = () => {
+    sendTrackEvent({
+      category: 'app_bar',
+      action: 'change_genome'
+    });
+  };
+
+  const trackTrackPanelToggled = (isOpened: boolean) => {
+    sendTrackEvent({
+      category: 'track_panel',
+      action: isOpened ? 'opened' : 'closed'
+    });
+  };
+
+  const trackTrackVisibilityToggled = (trackName: string, status: boolean) => {
+    sendTrackEvent({
+      category: 'regular_track_visibility',
+      action: 'turned_' + status ? 'on' : 'off',
+      label: trackName
+    });
+  };
+
+  const trackFocusTrackVisibilityToggled = (action: Status) => {
+    sendTrackEvent({
+      category: 'focus_track_visibility',
+      action: action === Status.SELECTED ? 'turned_on' : 'turned_off'
+    });
+  };
+
+  const trackTranscriptTrackVisibilityToggled = (
```

Comment:
Why did you decide to call it a transcript track? Was this a requirement from Lucy?

Reply:
No, it was not in the requirement. I thought differentiating the different types of tracks we interact with.

Reply:
What I don't like about this method is that the genome browser doesn't have such thing as a transcript track. It has a focus track; which, in case of a gene, can show one or more transcripts.